app/allocations/middleware/set-results.allocations.test.js

```diff
@@ -19,14 +20,24 @@ describe('Allocations middleware', function() {
     let res
     let req
     let next
+    let allocationsToTableStub
 
     beforeEach(function() {
+      allocationsToTableStub = sinon.stub().returnsArg(0)
       sinon.stub(allocationService, 'getActiveAllocations')
       sinon.stub(allocationService, 'getCancelledAllocations')
-      sinon.stub(presenters, 'allocationsToTable').returnsArg(0)
+      sinon
+        .stub(permissions, 'check')
+        .returns(false)
+        .withArgs('allocation:person:assign', ['allocation:person:assign'])
+        .returns(true)
+      sinon
+        .stub(presenters, 'allocationsToTableComponent')
+        .callsFake(() => allocationsToTableStub)
```

Comment:
Any reason to not use `returns` rather than `callsFake`?

Reply:
Ooo, didn't think of that...I'll give it a whirl!